src/machine.rs

```diff
@@ -1068,7 +1059,10 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
         kind: mir::RetagKind,
         place: &PlaceTy<'tcx, Provenance>,
     ) -> InterpResult<'tcx> {
-        if ecx.machine.stacked_borrows.is_some() { ecx.retag(kind, place) } else { Ok(()) }
+        match ecx.machine.borrow_tracker_method {
+            BorrowTrackerMethod::StackedBorrows => ecx.sb_retag(kind, place),
+            BorrowTrackerMethod::Off => Ok(()),
+        }
```

Comment:
Please move this `match` into `borrow_tracker/mod.rs`.
The machine file should never match on the `BorrowTrackerMethod`.